### DISCUSSION TITLE
.Net: I know the functions for the plugins expect an inmediate return value,  but what if I need somehow to create a plugin that depends on a bus

### DISCUSSION THREAD

**Discussion by DavidNorena** [microsoft/semantic-kernel#6073] [Q&A] [answered]
Sorry if this is not the place where I should be asking this, but what would be the approach if we need to develop a plugin that depends on information that comes from a service bus 

Like publish an event, and then just wait for the susbcribed event replies with an answer, is there an approach already in the kernel for this ?

Thanks

**Comment by sophialagerkranspandey** [comment, marked as answer]
Within native plugins you can add any native code including waiting for any asynchronous event to occur. Can you clarify that you're referring to native functions or are you trying to achieve this using OpenAPI calls?
